### PR TITLE
Alternative to source'ing the secrets file

### PR DESCRIPTION
**Pull request by jspaaks**
This PR could be an alternative to having to `source` the `rsd-secrets.env` in every terminal you do `docker-compose` work.

Check as follows:

1. new terminal
1. make sure you dont have a file called `.env`
1. `docker-compose config` should complain about missing required variables
1. add empty config file `touch .env` (it has to  be called that), `docker-compose config` should complain about variables
1. add whatever var it complains about, `docker-compose config` should then complain about the next missing variable
1. `rm .env`
1. Create a link named `.env` that points to your secrets file `ln -s rsd-secrets.env .env`; `docker-compose config` should return the full config and `docker-compose build && docker-compose up -d && docker-compose logs --follow` should work without the need to source every time; Also in new terminal windows you can just do `docker-compose exec database mongo rsd` or whatever without first source'ing.

Note the file *has* to be called `.env` (but it can be a link to another file)
Refs https://github.com/docker/compose/issues/4189#issuecomment-274915631

- ~~Still need to update the main README and the FAQs~~
- ~~Still need to check if the `BACKUP_CMD` works~~ should still work, `rsd-secrets.env.example` was changed and then changed back so still works like before
- ~~admin interface complains about `JWT_SECRET`~~ Looks like that was a browser cache problem

### CODE REVIEW

File: rsd-secrets.env.example [hunk 1]
```diff
@@ -1,7 +1,10 @@
+# Use of export below is ignored by docker-compose (specifically, python-dotenv)
+# but is useful for source'ing this file
+
 export AUTH_GITHUB_CLIENT_ID=changeme
 export AUTH_GITHUB_CLIENT_SECRET=changeme
 export AUTH_GITHUB_ORGANIZATION=changeme
-export BACKUP_CMD='xenon --version'
+export BACKUP_CMD=xenon --version
```
Comment (jspaaks) [inline rsd-secrets.env.example:7]:
```suggestion
export BACKUP_CMD='xenon --version'
```
Won't affect `docker-compose`, but without single quotes `export` doesn't work.

**Comment by eriktks** [comment]
Step 5 does not work for me. I do not see what I might be doing wrong:
```
$ git branch
* (HEAD detached at origin/required-vars)
  434-harvesting
  master
$ docker-compose build
ERROR: Invalid interpolation format for "environment" option in service "auth": "${JWT_SECRET:?Required variable JWT_SECRET not set or empty}"
$ echo "JWT_SECRET=test" > .env
$ docker-compose build
ERROR: Invalid interpolation format for "environment" option in service "auth": "${JWT_SECRET:?Required variable JWT_SECRET not set or empty}"
```

**Comment by jspaaks** [comment]
@sverhoeven do you think this approach is better than what we had?

**Comment by sverhoeven** [comment]
Yes, I frequently forgot to source rsd-secrets.env file. I would prefer to have rsd-secrets.env to be copied directly to .env if possible.

**Comment by eriktks** [comment]
My [problem](https://github.com/research-software-directory/research-software-directory/pull/468#issuecomment-707168436) disappeared after following a tip from Jurriaan to upgrade docker-compose

**Review by fdiblen — APPROVED** [review]
Thanks @jspaaks 
:cookie:

**Comment by jspaaks** [comment]
> My problem disappeared after following a tip from Jurriaan to upgrade docker-compose

Refs #427

**Comment by jspaaks** [comment]
Thanks reviewers!

**Pull request merged**